internal/handlers/organization.go

```diff
@@ -117,6 +116,26 @@ func (api *API) CreateOrganization(c *gin.Context) {
 		}
 		return
 	}
+
+	// Create a default security group for the organization
+	sg, err := api.createDefaultSecurityGroup(ctx, org.ID.String())
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.NewApiInternalError(err))
+		return
+	}
+
+	// Update the org with the new security group id, delete the security group if it fails
+	if err := api.updateOrganizationSecGroupId(ctx, sg.ID, org.ID); err != nil {
+		if res := api.db.Delete(&sg, "id = ?", sg.ID); res.Error != nil {
```

Comment:
My suggestion here was to move it into the existing transaction. That way, you don't have to add the manual rollback.

Reply:
yeah, i tried putting it in the same transaction, it didn't work. It breaks create orgs. No idea why.

Reply:
OK, I'll file a follow-up issue to look into it.

Reply:
https://github.com/nexodus-io/nexodus/issues/1131